### PR TITLE
The `GAE_APPLICATION_YAML_PATH` var is respected

### PR DESCRIPTION
**Pull request by DominicKramer**
Now if the `GAE_APPLICATION_YAML_PATH` environment variable is defined, its value will be used as the path to the application's `app.yaml` file.  Otherwise, if it is not defined, a default value of `app.yaml` will be used.

### CODE REVIEW

**Comment by DominicKramer** [comment]
@znewman01 I just wanted to double check with you that this is the expected behavior of using the `GAE_APPLICATION_YAML_PATH` environment variable.

**Comment by znewman01** [comment]
Logic looks consistent with my expectations.

Will tell you ASAP when we know that this is the way forward.

**Comment by DominicKramer** [comment]
@znewman01 Ok thanks.

**Comment by znewman01** [comment]
All systems go!

File: builder/steps/gen-dockerfile/contents/src/detect_setup.ts [hunk 1]
```diff
@@ -79,10 +84,13 @@ export interface Setup {
  */
 export async function detectSetup(
     logger: Logger, fsview: Reader&Locator): Promise<Setup> {
-  if (!(await fsview.exists(APP_YAML))) {
-    throw new Error(`The file ${APP_YAML} does not exist`);
+  const appYamlPath = process.env.GAE_APPLICATION_YAML_PATH ?
```
Comment (DominicKramer) [inline builder/steps/gen-dockerfile/contents/src/detect_setup.ts]: Fixed

File: builder/steps/gen-dockerfile/contents/src/detect_setup.ts [hunk 2]
```diff
@@ -49,6 +49,11 @@ export interface Setup {
    * dependencies and launch the app.
    */
   useYarn: boolean;
+  /**
+   * Specifies the path relative to the application's root directory that
+   * identifies the app.yaml file used to deploy the application.
```
Comment (DominicKramer) [inline builder/steps/gen-dockerfile/contents/src/detect_setup.ts]: Done.

**Pull request merged**